docs/tutorials/09-pilet-best-practices.md

```diff
@@ -329,6 +329,34 @@ export default ({ piral }) => (
 
 The crucial part is that we do not only place the registration of our testing components (in the example above the page) inside the environment-switch, but also the dependent modules. Above we only bundle the `PlaygroundPage` if we are indeed in the development environment. Production builds will never include this code.
 
+## TypeScript Declarations
+
+You might want to provide some TypeScript declarations for the extensions used by your pilet.
+
+When you use an extension you want to be sure that everything works, e.g.:
+
+```jsx
+app.registerPage('/example', () => (
+  <div>
+    <app.Extension name="foo" params={{ name: 'Bar' }} />
+  </div>
+));
+```
+
+Here, the `params` passed into the extension should autocomplete. The way to do that is by having the `PiralCustomExtensionSlotMap` interface extended. If your app shell is called `sample-piral` you can do that using:
```

Comment:
Can we specify where this should be configured (e.g., the file and/or project)?

Reply:
No need to specify. This is no where configured. It's just the types you write somewhere.

Reply:
(What would you write here to make it more clear? Just propose a change)

Reply:
```suggestion
Here, the `params` passed into the extension should autocomplete. To achieve this at the example of an app shell called `sample-piral`, create a type declaration within your pilet as follows:
```